core/View.php

```diff
@@ -330,15 +330,17 @@ public function setContentType($contentType)
      * _Note: setting this allows you to make sure the View **cannot** be
      * embedded in iframes. Learn more [here](https://developer.mozilla.org/en-US/docs/HTTP/X-Frame-Options)._
      *
-     * @param string $option ('deny' or 'sameorigin')
+     * @param string $option ('deny' or 'sameorigin' or 'allow-from URI')
      */
     public function setXFrameOptions($option = 'deny')
     {
         if ($option === 'deny' || $option === 'sameorigin') {
             $this->xFrameOptions = $option;
+            return;
         }
-        if ($option == 'allow') {
```

Comment:
@blueelvis I don't think  this code should be removed, since it's still used in https://github.com/piwik/piwik/blob/2.16.2/plugins/CoreAdminHome/OptOutManager.php#L203 and https://github.com/piwik/piwik/blob/master/plugins/Widgetize/Controller.php#L47-47